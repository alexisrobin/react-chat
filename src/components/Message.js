import React from 'react';
import './Message.scss';
import { Container, Row, Col } from 'reactstrap';

const Message = ({ message, senderName }) => {

    const convertISOStringToReadableDateString = ISOString =>
        new Intl.DateTimeFormat('en-US', { hour: '2-digit', minute: '2-digit', year: 'numeric', month: 'short', day: '2-digit' }).format(new Date(Date.parse(ISOString)));

    return (
        <Container>
            <Row className="message-content rounded p-2">
                {message.content}
            </Row>
            <Row className="message-timestamp">
                <Col className="text-right">
                    sent by {senderName} - {convertISOStringToReadableDateString(message.timestamp)}
                </Col>
            </Row>
        </Container>
    );
}

export default Message;