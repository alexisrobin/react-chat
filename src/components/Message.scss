.message-content {
    background-color: #ddd;
}

.message-timestamp {
    color: #aaa;
    font-size: 0.9rem;
    font-style: italic;
}