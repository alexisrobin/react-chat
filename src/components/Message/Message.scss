@import '../../const.scss';

.message-sent {
    .message-content {
        background-color: $primary;
        color: white;
    }
    .message-timestamp {
        text-align: right;
    }
}

.message-received {
    .message-content {
        background-color: #ddd;
        color: black;
    }
    .message-timestamp {
        text-align: left;
    }
}

.message-timestamp {
    color: #aaa;
    font-size: 0.9rem;
    font-style: italic;
}

.message-unreadable {
    text-align: center;
    font-style: italic;
}