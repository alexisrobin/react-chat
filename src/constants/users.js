export const users = [
    {
        id: 1,
        name: "Chat A"
    },
    {
        id: 2,
        name: "Chat B"
    }
]