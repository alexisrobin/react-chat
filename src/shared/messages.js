export const messages = [
    {
        "id": 0,
        "senderId": 1,
        "content": "Mollit minim adipisicing tempor reprehenderit. Eu aliquip magna aliquip eiusmod. Aliquip anim exercitation elit eiusmod aute magna amet anim proident. Tempor ut enim ullamco elit veniam cillum cupidatat sint. Aliquip pariatur culpa amet non aliquip anim non dolore elit. Incididunt amet Lorem occaecat cupidatat officia dolor deserunt tempor nisi aute occaecat non ut. Cillum duis exercitation adipisicing reprehenderit quis ipsum veniam id commodo enim qui amet duis ut.\r\n",
        "timestamp": "2016-10-30T10:28:05"
    },
    {
        "id": 1,
        "senderId": 2,
        "content": "Id velit aute elit ipsum anim officia. Non incididunt minim ipsum deserunt nulla qui labore ex. Dolore aliquip et amet aute occaecat deserunt deserunt Lorem mollit do irure. Consectetur eu adipisicing excepteur in aliquip quis do ut.\r\n",
        "timestamp": "2015-02-24T03:42:08"
    },
    {
        "id": 2,
        "senderId": 1,
        "content": "Exercitation ad nostrud et proident reprehenderit enim esse in qui voluptate ea. Aliquip mollit esse sunt quis consequat exercitation consectetur sunt laboris nisi ipsum. Mollit occaecat deserunt occaecat veniam exercitation labore adipisicing aliquip dolore culpa officia nostrud ea. Lorem commodo ad nisi consectetur ad nostrud.\r\n",
        "timestamp": "2016-04-29T05:57:31"
    },
    {
        "id": 3,
        "senderId": 1,
        "content": "In ut sint ut ex laborum magna dolore do est. Deserunt dolore minim consectetur nostrud dolor. Fugiat tempor sit aliquip adipisicing ea ex mollit mollit commodo Lorem culpa in ad. Deserunt consectetur deserunt sit enim ea. Cupidatat exercitation irure irure eu. Deserunt incididunt minim magna in qui proident reprehenderit Lorem enim proident id excepteur excepteur. Occaecat ullamco est dolor commodo eu est dolore.\r\n",
        "timestamp": "2016-02-06T07:55:13"
    },
    {
        "id": 4,
        "senderId": 1,
        "content": "Do deserunt labore ullamco aute pariatur quis elit ad excepteur ut ex magna. Ullamco in eiusmod consequat proident occaecat do nisi commodo. Esse ex consequat aliquip pariatur voluptate est magna. Cillum consequat excepteur aliqua occaecat ea id anim eiusmod laborum sit.\r\n",
        "timestamp": "2015-01-05T05:24:06"
    },
    {
        "id": 5,
        "senderId": 2,
        "content": "Exercitation aliqua sit ullamco mollit fugiat elit do velit ex voluptate officia cupidatat aliqua. Laborum enim eiusmod occaecat laboris eiusmod exercitation non cillum sint do. Ipsum cillum est duis ad non tempor consequat Lorem.\r\n",
        "timestamp": "2016-06-25T11:16:49"
    },
    {
        "id": 6,
        "senderId": 1,
        "content": "Quis ea ut minim qui officia excepteur cillum commodo adipisicing veniam duis. Do eu ex exercitation culpa sint magna voluptate. Lorem mollit nostrud minim magna ea non duis incididunt nulla. Nostrud ea velit duis exercitation. Lorem ad reprehenderit ea ea non excepteur sint magna culpa laborum ex ad duis. Est est ut eiusmod officia duis aliqua proident et aliqua do adipisicing culpa.\r\n",
        "timestamp": "2016-01-01T10:05:01"
    },
    {
        "id": 7,
        "senderId": 2,
        "content": "Nulla pariatur nulla cupidatat duis elit dolore tempor velit excepteur. Officia ipsum irure exercitation aliqua enim elit. Cupidatat deserunt ad culpa excepteur voluptate.\r\n",
        "timestamp": "2016-07-04T11:24:32"
    },
    {
        "id": 8,
        "senderId": 2,
        "content": "Non exercitation velit veniam aliqua ut anim. Voluptate id quis ea officia eiusmod et do nisi ut fugiat tempor. Fugiat occaecat reprehenderit et ea aute eu proident laboris. Aliqua consequat adipisicing deserunt labore aliquip fugiat ad irure eu fugiat do deserunt. Quis exercitation nulla sit id labore.\r\n",
        "timestamp": "2018-03-30T08:28:05"
    },
    {
        "id": 9,
        "senderId": 1,
        "content": "Tempor in culpa veniam nostrud. Laboris aliqua ea ea anim labore proident et duis id. Nulla do mollit minim sunt. Sunt officia aliqua minim anim. Anim commodo id dolor elit est quis Lorem nostrud fugiat reprehenderit anim. Proident nulla laborum do sit velit sint exercitation ullamco deserunt laborum et in voluptate consectetur.\r\n",
        "timestamp": "2014-02-09T08:32:25"
    },
    {
        "id": 10,
        "senderId": 1,
        "content": "Dolore est cupidatat id ullamco. Eiusmod eu commodo mollit irure qui velit eiusmod irure laborum duis irure ullamco cillum aliqua. Ex irure fugiat aliqua cupidatat id reprehenderit ipsum nostrud qui cillum elit.\r\n",
        "timestamp": "2017-04-16T09:37:54"
    },
    {
        "id": 11,
        "senderId": 1,
        "content": "Quis exercitation deserunt anim amet. Deserunt excepteur id laboris duis deserunt dolore esse ad duis cupidatat labore ex eiusmod. Ullamco excepteur Lorem quis adipisicing aliqua nisi anim officia consequat irure incididunt. Do mollit adipisicing ipsum et. Amet aliqua incididunt ex officia mollit est laboris occaecat excepteur enim ea aliquip tempor. Ea ut sint adipisicing deserunt cillum fugiat commodo proident tempor aliquip adipisicing incididunt sint.\r\n",
        "timestamp": "2017-12-11T11:11:40"
    },
    {
        "id": 12,
        "senderId": 2,
        "content": "Nisi ad excepteur minim excepteur nostrud in non nisi labore ea deserunt culpa sit deserunt. Do irure irure aute quis veniam ad voluptate enim. Excepteur do irure minim magna ea anim nisi et laborum id irure deserunt esse. Irure veniam eiusmod consequat voluptate ad quis do consectetur cillum irure exercitation consequat. Dolore esse occaecat amet in.\r\n",
        "timestamp": "2015-01-22T09:12:28"
    },
    {
        "id": 13,
        "senderId": 2,
        "content": "Ullamco culpa dolor est reprehenderit ullamco exercitation. Incididunt velit id irure proident mollit ut adipisicing Lorem anim adipisicing non cupidatat. Occaecat laboris consequat consequat esse ad consequat laboris ex sunt nostrud adipisicing anim officia. Aliquip amet id duis est aute. Sint cupidatat amet non ut excepteur aliqua magna aute ipsum deserunt amet est nostrud.\r\n",
        "timestamp": "2015-08-20T12:17:56"
    },
    {
        "id": 14,
        "senderId": 1,
        "content": "Qui et id tempor tempor adipisicing ad velit aute. Sit eiusmod non voluptate ut amet sunt. Reprehenderit magna ea voluptate tempor dolore veniam ut enim deserunt nulla. Tempor minim nostrud do sit Lorem non fugiat eu sint ullamco occaecat reprehenderit. Ad id sint pariatur commodo do. Dolore aliqua nostrud incididunt amet fugiat.\r\n",
        "timestamp": "2018-01-27T12:15:51"
    },
    {
        "id": 15,
        "senderId": 2,
        "content": "Nisi Lorem nostrud nulla incididunt laborum velit incididunt fugiat ullamco eu minim et duis culpa. Quis aute ad dolor do aliquip tempor aute nostrud. Sit labore nisi ipsum magna eu voluptate sint eiusmod cupidatat proident et mollit consectetur.\r\n",
        "timestamp": "2015-05-02T11:58:37"
    }
]