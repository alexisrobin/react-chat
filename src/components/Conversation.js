import React from 'react';
import './Conversation.scss';
import { Media } from 'reactstrap';
import Message from './Message';


const Conversation = ({ messages, users }) => {
    const currentUser = 1;
    // Sorting messages by timestamp
    messages = messages.sort(function(a,b){
        return new Date(a.timestamp) - new Date(b.timestamp);
    });
    return (
        <div className="conversation">
            <Media list className="p-0">
                {messages.map(message => {
                    return (
                        <Media tag="li"
                            key={message.id}
                            className={message.senderId === currentUser ? "justify-content-end" : "justify-content-start"}>
                            <div>
                                <Message
                                    message={message}
                                    senderName={users.filter(user => user.id === message.senderId).map(user => user.name)} />
                            </div>
                        </Media>
                    )
                })}
            </Media>
        </div>
    )
}

export default Conversation;