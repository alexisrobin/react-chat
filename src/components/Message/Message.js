import React from 'react';
import './Message.scss';
import { Container, Row, Col } from 'reactstrap';
import { FadeTransform, } from 'react-animation-components';


const convertDateFromISOStringToReadableString = ISOString =>
    new Intl.DateTimeFormat('en-US', { hour: '2-digit', minute: '2-digit', year: 'numeric', month: 'short', day: '2-digit' }).format(new Date(Date.parse(ISOString)));

/**
 * Basic chat's message adapting it's view depending of its nature (sent, received).
 */
const Message = ({ message, sender, currentUserId }) => {
    if (!message) {
        return (
            <div className="message-unreadable">
                Unreadable message
            </div>
        );
    }
    else
        return (
            <FadeTransform
                in
                transformProps={
                    sender.id === currentUserId ?
                        {
                            enterTransform: 'translateX(0px)',
                            exitTransform: 'translateX(-100px)'
                        } :
                        {
                            enterTransform: 'translateX(0px)',
                            exitTransform: 'translateX(100px)'
                        }
                }>
                <Container className={sender.id === currentUserId ? "message-sent" : "message-received"}>
                    <Row className="message-content rounded p-2">
                        {message.content}
                    </Row>
                    <Row className="message-timestamp">
                        <Col>
                            sent by {sender.name} - {convertDateFromISOStringToReadableString(message.timestamp)}
                        </Col>
                    </Row>
                </Container>
            </FadeTransform>
        );
}

export default Message;
export { convertDateFromISOStringToReadableString };