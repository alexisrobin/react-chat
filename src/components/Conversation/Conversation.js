import React, { Component } from 'react';
import './Conversation.scss';
import { Media } from 'reactstrap';
import Message from '../Message';

// Sort array by timestamp property;
const sortByTimestamp = array => array.sort((a, b) => new Date(a.timestamp) - new Date(b.timestamp));

/**
 * Conversation between users.
 */
class Conversation extends Component {
    
    /**
     * Scroll to conversation latest message.
     */
    scrollToLatestMessage = () => {
        if(!!this.conversationBottom)
            this.conversationBottom.parentNode.scrollTop = this.conversationBottom.offsetTop;
    }

    /**
     * Ensure to scroll to the latest message when component did mount.
     */
    componentDidMount() {
        this.scrollToLatestMessage();
    }

    /**
     * Ensure to scroll to the latest message when component did update.
     */
    componentDidUpdate() {
        this.scrollToLatestMessage();
    }

    render() {
        return (
            <div className="conversation">
                <Media list className="p-1">
                    {sortByTimestamp(this.props.messages).map((message) => {
                        return (
                            <Media tag="li"
                                key={message.id}
                                className={message.senderId === this.props.currentUserId ? "justify-content-end" : "justify-content-start"}>
                                <div className="p-1">
                                    <Message
                                        message={message}
                                        sender={this.props.users.filter(user => user.id === message.senderId).shift()}
                                        currentUserId={this.props.currentUserId} />
                                </div>
                            </Media>
                        )
                    })}
                </Media>
                <div ref={(el) => { this.conversationBottom = el; }}></div>
            </div>
        )
    }
}

export default Conversation;