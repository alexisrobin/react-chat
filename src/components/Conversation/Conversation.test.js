import React from 'react';
import { expect } from 'chai'
import { shallow, render } from 'enzyme'
import Conversation from './Conversation';
import Message from '../Message/Message';
import { messages } from '../../constants/messages'
import { users } from '../../constants/users'

describe('<Conversation />', () => {
    it('should match its reference snapshot', () => {
        const wrapper = render(
            <Conversation
                messages={messages}
                users={users}
                currentUserId={1} />
        );

        expect(wrapper).to.matchSnapshot()
    })

    it('should render all messages', () => {
        const wrapper = shallow(
            <Conversation
                messages={messages}
                users={users} />
        );

        expect(wrapper.find(Message)).to.have.length(messages.length);
    })
})