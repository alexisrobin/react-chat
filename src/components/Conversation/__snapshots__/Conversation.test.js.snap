// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Conversation /> should match its reference snapshot 1`] = `
<div
  class="conversation"
>
  <ul
    class="p-1 media-list"
  >
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Tempor in culpa veniam nostrud. Laboris aliqua ea ea anim labore proident et duis id. Nulla do mollit minim sunt. Sunt officia aliqua minim anim. Anim commodo id dolor elit est quis Lorem nostrud fugiat reprehenderit anim. Proident nulla laborum do sit velit sint exercitation ullamco deserunt laborum et in voluptate consectetur.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Feb 09, 2014, 8:32 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Do deserunt labore ullamco aute pariatur quis elit ad excepteur ut ex magna. Ullamco in eiusmod consequat proident occaecat do nisi commodo. Esse ex consequat aliquip pariatur voluptate est magna. Cillum consequat excepteur aliqua occaecat ea id anim eiusmod laborum sit.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Jan 05, 2015, 5:24 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-start media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-received container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Nisi ad excepteur minim excepteur nostrud in non nisi labore ea deserunt culpa sit deserunt. Do irure irure aute quis veniam ad voluptate enim. Excepteur do irure minim magna ea anim nisi et laborum id irure deserunt esse. Irure veniam eiusmod consequat voluptate ad quis do consectetur cillum irure exercitation consequat. Dolore esse occaecat amet in.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat B - Jan 22, 2015, 9:12 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-start media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-received container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Id velit aute elit ipsum anim officia. Non incididunt minim ipsum deserunt nulla qui labore ex. Dolore aliquip et amet aute occaecat deserunt deserunt Lorem mollit do irure. Consectetur eu adipisicing excepteur in aliquip quis do ut.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat B - Feb 24, 2015, 3:42 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-start media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-received container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Nisi Lorem nostrud nulla incididunt laborum velit incididunt fugiat ullamco eu minim et duis culpa. Quis aute ad dolor do aliquip tempor aute nostrud. Sit labore nisi ipsum magna eu voluptate sint eiusmod cupidatat proident et mollit consectetur.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat B - May 02, 2015, 11:58 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-start media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-received container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Ullamco culpa dolor est reprehenderit ullamco exercitation. Incididunt velit id irure proident mollit ut adipisicing Lorem anim adipisicing non cupidatat. Occaecat laboris consequat consequat esse ad consequat laboris ex sunt nostrud adipisicing anim officia. Aliquip amet id duis est aute. Sint cupidatat amet non ut excepteur aliqua magna aute ipsum deserunt amet est nostrud.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat B - Aug 20, 2015, 12:17 PM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Quis ea ut minim qui officia excepteur cillum commodo adipisicing veniam duis. Do eu ex exercitation culpa sint magna voluptate. Lorem mollit nostrud minim magna ea non duis incididunt nulla. Nostrud ea velit duis exercitation. Lorem ad reprehenderit ea ea non excepteur sint magna culpa laborum ex ad duis. Est est ut eiusmod officia duis aliqua proident et aliqua do adipisicing culpa.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Jan 01, 2016, 10:05 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                In ut sint ut ex laborum magna dolore do est. Deserunt dolore minim consectetur nostrud dolor. Fugiat tempor sit aliquip adipisicing ea ex mollit mollit commodo Lorem culpa in ad. Deserunt consectetur deserunt sit enim ea. Cupidatat exercitation irure irure eu. Deserunt incididunt minim magna in qui proident reprehenderit Lorem enim proident id excepteur excepteur. Occaecat ullamco est dolor commodo eu est dolore.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Feb 06, 2016, 7:55 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Exercitation ad nostrud et proident reprehenderit enim esse in qui voluptate ea. Aliquip mollit esse sunt quis consequat exercitation consectetur sunt laboris nisi ipsum. Mollit occaecat deserunt occaecat veniam exercitation labore adipisicing aliquip dolore culpa officia nostrud ea. Lorem commodo ad nisi consectetur ad nostrud.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Apr 29, 2016, 5:57 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-start media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-received container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Exercitation aliqua sit ullamco mollit fugiat elit do velit ex voluptate officia cupidatat aliqua. Laborum enim eiusmod occaecat laboris eiusmod exercitation non cillum sint do. Ipsum cillum est duis ad non tempor consequat Lorem.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat B - Jun 25, 2016, 11:16 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-start media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-received container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Nulla pariatur nulla cupidatat duis elit dolore tempor velit excepteur. Officia ipsum irure exercitation aliqua enim elit. Cupidatat deserunt ad culpa excepteur voluptate.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat B - Jul 04, 2016, 11:24 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Mollit minim adipisicing tempor reprehenderit. Eu aliquip magna aliquip eiusmod. Aliquip anim exercitation elit eiusmod aute magna amet anim proident. Tempor ut enim ullamco elit veniam cillum cupidatat sint. Aliquip pariatur culpa amet non aliquip anim non dolore elit. Incididunt amet Lorem occaecat cupidatat officia dolor deserunt tempor nisi aute occaecat non ut. Cillum duis exercitation adipisicing reprehenderit quis ipsum veniam id commodo enim qui amet duis ut.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Oct 30, 2016, 10:28 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Dolore est cupidatat id ullamco. Eiusmod eu commodo mollit irure qui velit eiusmod irure laborum duis irure ullamco cillum aliqua. Ex irure fugiat aliqua cupidatat id reprehenderit ipsum nostrud qui cillum elit.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Apr 16, 2017, 9:37 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Quis exercitation deserunt anim amet. Deserunt excepteur id laboris duis deserunt dolore esse ad duis cupidatat labore ex eiusmod. Ullamco excepteur Lorem quis adipisicing aliqua nisi anim officia consequat irure incididunt. Do mollit adipisicing ipsum et. Amet aliqua incididunt ex officia mollit est laboris occaecat excepteur enim ea aliquip tempor. Ea ut sint adipisicing deserunt cillum fugiat commodo proident tempor aliquip adipisicing incididunt sint.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Dec 11, 2017, 11:11 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-end media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(-100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-sent container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Qui et id tempor tempor adipisicing ad velit aute. Sit eiusmod non voluptate ut amet sunt. Reprehenderit magna ea voluptate tempor dolore veniam ut enim deserunt nulla. Tempor minim nostrud do sit Lorem non fugiat eu sint ullamco occaecat reprehenderit. Ad id sint pariatur commodo do. Dolore aliqua nostrud incididunt amet fugiat.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat A - Jan 27, 2018, 12:15 PM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
    <li
      class="justify-content-start media"
    >
      <div
        class="p-1"
      >
        <div
          style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;opacity:0;transition-property:opacity;will-change:opacity"
        >
          <div
            style="transition-delay:0ms;transition-duration:500ms;transition-timing-function:ease;transform:translateX(100px);transition-property:transform;will-change:transform"
          >
            <div
              class="message-received container"
            >
              <div
                class="message-content rounded p-2 row"
              >
                Non exercitation velit veniam aliqua ut anim. Voluptate id quis ea officia eiusmod et do nisi ut fugiat tempor. Fugiat occaecat reprehenderit et ea aute eu proident laboris. Aliqua consequat adipisicing deserunt labore aliquip fugiat ad irure eu fugiat do deserunt. Quis exercitation nulla sit id labore.

              </div>
              <div
                class="message-timestamp row"
              >
                <div
                  class="col"
                >
                  sent by Chat B - Mar 30, 2018, 8:28 AM
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </li>
  </ul>
  <div
    style="float:left;clear:both"
  />
</div>
`;
